### PR TITLE
[Heartbeat]: add mappings for performance metrics

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fix https://github.com/elastic/beats/issues/26122
+ Mapping for the Performance metrics captured by the Synthetics agent. 
+ Counterpart of https://github.com/elastic/synthetics/pull/194

### CODE REVIEW

File: heartbeat/_meta/fields.common.yml [hunk 1]
```diff
@@ -174,6 +174,45 @@
...
+              type: group
+              fields:
+                - name: name
+                  type: keyword
+                - name: type
+                  type: text
+                - name: start
+                  type: long
+                  description: >
+                    offset of time relative to journey start
```
Comment (andrewvc) [inline heartbeat/_meta/fields.common.yml:201]: Can we indicate the unit of measure? I believe it's seconds as a float?

File: heartbeat/_meta/fields.common.yml [hunk 2]
```diff
@@ -174,6 +174,45 @@
...
+                - name: end
+                  type: long
+                  description: >
+                    offset of time relative to journey start
```
Comment (andrewvc) [inline heartbeat/_meta/fields.common.yml:205]: Can we indicate the unit of measure? I believe it's seconds as a float?

File: heartbeat/_meta/fields.common.yml [hunk 3]
```diff
@@ -174,6 +174,45 @@
...
+                - name: start
+                  type: long
+                  description: >
+                    duration relative to journey start in milliseconds
+            - name: user_timing
+              type: group
+              fields:
+                - name: name
+                  type: keyword
+                - name: type
```
Comment (andrewvc) [inline heartbeat/_meta/fields.common.yml:196]: Can we document the `type`? It doesn't need to be exhaustive. I see `mark` in our test fixtures in synthetics, what are the other types?

**Review by andrewvc — COMMENTED** [review]
Could use some improvements on the docs side, but otherwise good

**Review by andrewvc — APPROVED** [review]
LGTM

**Comment by vigneshshanmugam** [comment]
@Mergifyio update

**Comment by vigneshshanmugam** [comment]
@Mergifyio backport 7.x

**Pull request merged**